src/keyDerivation_test.c

```diff
@@ -0,0 +1,141 @@
+#include "keyDerivation.h"
+#include "test_utils.h"
+#include "hex_utils.h"
+#include "stream.h"
+#include <string.h>
+#include "errors.h"
+
+#define ARRAYLEN(array) (sizeof(array) / sizeof(array[0]))
+
+void testPrivateKeyDerivation()
+{
+#define TESTCASE(path_, pathLength_, expected_) \
+	{ \
+		uint8_t chainCode[32]; \
+		cx_ecfp_256_extended_private_key_t privateKey; \
+		derivePrivateKey(path_, pathLength_, chainCode, &privateKey); \
+		stream_t s; \
+		stream_init(&s); \
+		stream_appendFromHexString(&s, expected_); \
+		const uint8_t* expectedBuffer = stream_head(&s); \
+		EXPECT_EQ_BYTES(expectedBuffer, privateKey.d, 64); \
+	}
+
+	{
+		uint32_t path1[] = { HARDENED_BIP32 + 44, HARDENED_BIP32 + 1815, HARDENED_BIP32 + 1};
```

Comment:
Technical note: if you wrapped every testcase in separate {} scope, there wouldn't be a need for numbering path1, path2, ... etc. I am not sure if it is worth a level of indent though...

Reply:
Yep I know, but currently I would rather use the scope for logical separation, e.g. one scope for non-throwing code, another scope for throwing code. Or one scope for small paths, another scope for long paths etc.

Reply:
Ok, makes sense